public/app/core/components/ForgottenPassword/ChangePasswordPage.test.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { act, fireEvent, render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { getRouteComponentProps } from 'app/core/navigation/__mocks__/routeProps';
+
+import { ChangePasswordPage, Props } from './ChangePasswordPage';
+
+const postMock = jest.fn();
+jest.mock('@grafana/runtime', () => ({
+  getBackendSrv: () => ({
+    post: postMock,
+  }),
+}));
+
+jest.mock('app/core/config', () => {
+  return {
+    loginError: false,
+    buildInfo: {
+      version: 'v1.0',
+      commit: '1',
+      env: 'production',
+      edition: 'Open Source',
+      isEnterprise: false,
+    },
+    licenseInfo: {
+      stateInfo: '',
+      licenseUrl: '',
+    },
+    appSubUrl: '',
+  };
+});
+const props: Props = {
+  ...getRouteComponentProps({
+    queryParams: { code: 'some code' },
+  }),
+};
+
+describe('ChangePassword Page', () => {
+  it('renders correctly', () => {
+    render(<ChangePasswordPage {...props} />);
+
+    expect(screen.getByLabelText('New password')).toBeInTheDocument();
+    expect(screen.getByLabelText('Confirm new password')).toBeInTheDocument();
+
+    expect(screen.getByRole('button', { name: 'Submit' })).toBeInTheDocument();
+  });
+  it('should pass validation checks for password and confirm password field', async () => {
+    render(<ChangePasswordPage {...props} />);
+
+    fireEvent.click(screen.getByRole('button', { name: 'Submit' }));
+    expect(await screen.findByText('New password is required')).toBeInTheDocument();
+    expect(await screen.findByText('Confirmed password is required')).toBeInTheDocument();
```

Comment:
```suggestion
    expect(screen.getByText('Confirmed password is required')).toBeInTheDocument();
```
You don't need to `await` the second time since by then all async actions have already been resolved.

Reply:
addressed